### PR TITLE
feat(cli): support multiple formats and outputs at once

### LINKED ISSUE #1970: Generate multiple output formats with one cli call
**User story.**
As a spectral user, I would like to be able to generate lint reports in different formats with a single cli call without the need or lining the same spec multiple times (it can be time-consuming for huge specs).

**Describe the solution you'd like**
An option is to change `--output` and `--format` cli flags to accept a comma separated list of values so that a user can decide to generate more than one report and save it into different files. In particular

```bash
# as today, stylish report on stdout 
spectral lint --ruleset ruleset.yml openapi.json

# as today, json report on stdout 
spectral lint --ruleset ruleset.yml -f json openapi.json

# json report on file report.json, stylish report on file report.txt
spectral lint --ruleset ruleset.yml -f json,stylish -o report.json,report.txt openapi.json

# json report on file report.json, stylish report on stdout
spectral lint --ruleset ruleset.yml -f json,stylish -o report.json,stdout openapi.json

# TBD error or only json report on stdout
spectral lint --ruleset ruleset.yml -f json,stylish openapi.json
```

I can create a PR if you think is worth adding the feature

### PR DESCRIPTION
**Pull request by P0lip**
Closes https://github.com/stoplightio/spectral/issues/1970
Branched off https://github.com/stoplightio/spectral/pull/2011

**Checklist**

- [x] Tests added / updated
- [ ] Docs added / updated

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

**Additional context**

I didn't dare to bother @albertored, so I branched off Alberto's great PR and applied 2 suggestions @fgreinacher and @albertored made there:
- https://github.com/stoplightio/spectral/pull/2011#discussion_r790256353
- https://github.com/stoplightio/spectral/pull/2011#discussion_r790510088

If we all like it, we could merge it back to https://github.com/stoplightio/spectral/pull/2011 and then merge https://github.com/stoplightio/spectral/pull/2011 into develop.

### CODE REVIEW

File: packages/cli/src/commands/lint.ts [hunk 1]
```diff
@@ -34,15 +35,42 @@ const lintCommand: CommandModule = {
...
+
+          throw new TypeError('Output must be either string or unspecified when a single format is specified');
+        }
+
+        if (!isPlainObject(output)) {
+          throw new TypeError('Multiple outputs have to be provided when more than a single format is specified');
+        }
+
+        const keys = Object.keys(output);
+        if (format.length !== keys.length) {
```
Comment (albertored) [inline packages/cli/src/commands/lint.ts:77]: @P0lip looks great but if I understand correctly there is no way to have something like
`--format json --format html --output.json out.json`
to have one (or more) output(s) to file and one to stdout, right?

Comment (fgreinacher) [reply packages/cli/src/commands/lint.ts:77]: Would be really nice to retain the `stdout` default

**Comment by fgreinacher** [comment]
Looks really cool and simple. Just one wish from my side, otherwise good to go IMO!

File: packages/cli/src/commands/lint.ts (hunk 1 above)
Comment (P0lip) [reply packages/cli/src/commands/lint.ts:77]: yeah, no way to do that.
I can add it back though

**Comment by P0lip** [comment]
@fgreinacher @albertored done. What do you think?

File: packages/cli/src/commands/lint.ts [hunk 2]
```diff
@@ -36,9 +36,21 @@ const lintCommand: CommandModule = {
...
-          argv.output = { [format]: argv.output };
+        const formats = argv.format as string[] & { 0: string };
+        if (argv.output === void 0) {
+          argv.output = { [formats[0]]: '<stdout>' };
+        } else if (typeof argv.output === 'string') {
+          argv.output = { [formats[0]]: argv.output };
+        } else {
+          const output = argv.output as Dictionary<unknown>;
+          if (Object.keys(output).length + 1 >= formats.length) {
+            return;
```
Comment (fgreinacher) [inline packages/cli/src/commands/lint.ts:47]: Should we maybe report an error in this case?

**Comment by fgreinacher** [comment]
> @fgreinacher @albertored done. What do you think?

@P0lip Just one minor thing from my side, otherwise looking great!

**Comment by albertored** [comment]
> @fgreinacher @albertored done. What do you think?

@P0lip looks great to me!

File: packages/cli/src/commands/lint.ts (hunk 2 above)
Comment (P0lip) [reply packages/cli/src/commands/lint.ts:47]: This should be caught by https://github.com/stoplightio/spectral/blob/fdcdabd99e8cf1a87935ea4cd02722ac36e219e1/packages/cli/src/commands/lint.ts#L77

**Comment by P0lip** [comment]
@albertored would you mind cherry-picking these commits onto your branch or should I just merge to develop?
I'm afraid merging into a branch living in a fork doesn't seem to be an option 😞

**Comment by albertored** [comment]
@P0lip simply merge this one, no problem for me

**Comment by P0lip** [comment]
👍 it's in then :tada:
It's going to be replaced tomorrow at around 1pm CET

**Pull request merged**